dali/test/python/test_operator_random_object_bbox.py

```diff
@@ -0,0 +1,364 @@
+            class_labels = classes_out.at(i)
+            if background is not None or classes is None:
+                background_label = background_out.at(i)
+            else:
+                background_label = 0 if 0 not in class_labels else np.min(class_labels)-1
+
+            label = box_class_labels[i]
+            if classes is not None:
+                assert label == background_label or label in list(class_labels)
+
+            is_foreground = label != background_label
+            cls_boxes = class_boxes(in_tensor, label if is_foreground else None)
+
+            if is_foreground:
+                ref_boxes = cls_boxes
+                if threshold is not None:
+                    extent = box_extent(boxes[i])
+                    thr = threshold_out.at(i)
+                    assert np.all(extent >= thr)
+                    ref_boxes = list(filter(lambda box: np.all(box_extent(box) >= thr), cls_boxes))
+                if k_largest is not None:
+                    assert box_in_k_largest(ref_boxes, boxes[i], k_largest)
+            assert contains_box(cls_boxes, boxes[i])
+
+
+def test_random_object_bbox_with_class():
+    np.random.seed(12345)
+    types = [np.int8, np.uint8, np.int16, np.uint16, np.int32, np.uint32]
+
+    formats = [None, "anchor_shape", "start_end", "box"]
+    fmt = 0
+    for bg in [None, 0, -1, 5, fn.random.uniform(range=(-5, 10), dtype=dali.types.INT32, seed=12321)]:
+        if bg is None or isinstance(bg, int):
+            class_opt = [None, [0], [1], [2,4,5,7]]
+            for x in class_opt:
+                if isinstance(x, list) and bg in x:
+                    x.remove(bg)
+            if [] in class_opt: class_opt.remove([])
+            # putting this in the list interefered with remove
+            class_opt.append(random_classes(0 if bg is None else bg))
+        else:
+            class_opt = [None]
+        for classes in class_opt:
+            if classes is None:
+                weights_opt = [None, [1], [0.5, 1, 0.1, 0.2], random_weights()]
+            elif isinstance(classes, list):
+                weights_opt = [None, list(range(1, 1+len(classes)))]
+            else:
+                weights_opt = [None]
+
+            for weights in weights_opt:
+                ndim = np.random.randint(1, 5)
+
+                threshold_opt = [None, 3, list(range(1, 1+ndim)), fn.random.uniform(range=(1,5), shape=[ndim], dtype=dali.types.INT32, seed=13231)]
+                threshold = threshold_opt[np.random.randint(len(threshold_opt))]
+                k_largest_opt = [None, 1, 2, 5]
+                k_largest = k_largest_opt[np.random.randint(len(k_largest_opt))]
+
+                fg_prob_opt = [None, 0.1, 0.7, fn.random.uniform(range=(0,1), seed=1515)]
+                fg_prob = fg_prob_opt[np.random.randint(len(fg_prob_opt))]
+
+                format = formats[fmt]
+                fmt = (fmt + 1) % len(formats)
+                dtype = types[np.random.randint(0, len(types))]
```

Comment:
Use `np.random.choice(types)`

Reply:
Actually, now numpy complain that random choice is deprecated for non-numeric types... I'd leave it as-is and fix it in the parallel PR...?